snuba/migrations/snuba_migrations/outcomes/0003_outcomes_add_category_and_quantity.py

```diff
@@ -0,0 +1,118 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import Column, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations
+
+
+class Migration(migration.MultiStepMigration):
+    """
+    Adds quantity and category columns to outcomes. updates hourly table to support
+    category as a new dimension and quantity as a new measure.
+    """
+
+    blocking = False
+
+    def forwards_local(self) -> Sequence[operations.Operation]:
+        return [
+            operations.AddColumn(
+                storage_set=StorageSetKey.OUTCOMES,
+                table_name="outcomes_raw_local",
+                column=Column("quantity", UInt(32)),
+                after="reason",
+            ),
+            operations.AddColumn(
+                storage_set=StorageSetKey.OUTCOMES,
+                table_name="outcomes_raw_local",
+                column=Column("category", UInt(8)),
+                after="timestamp",
+            ),
+            operations.AddColumn(
+                storage_set=StorageSetKey.OUTCOMES,
+                table_name="outcomes_hourly_local",
+                column=Column("quantity", UInt(64)),
```

Comment:
Why is quantity UInt64 on the aggregate table and UInt32 in the raw table ?

Reply:
@fpacifici the max limit of an attachment is 20 MiB, and i don't think we'd ever aggregate more than 2^32 events in one outcome, for for a raw outcome i think that makes sense. someone sending lots of attachments could in theory go over 4GB attachments in an hour.

Reply:
ok